internal/kubelet/kubeconfig.go

```diff
@@ -117,11 +121,23 @@ func generateKubeconfig(cfg *api.NodeConfig) ([]byte, error) {
 // GetKubeClientFromKubeConfig gets kubernetes client from kubeconfig on the disk
 func GetKubeClientFromKubeConfig() (kubernetes.Interface, error) {
 	// Use the current context in the kubeconfig file
-	config, err := clientcmd.BuildConfigFromFlags("", KubeconfigPath())
+	config, err := clientcmd.LoadFromFile(KubeconfigPath())
 	if err != nil {
-		return nil, errors.Wrap(err, "failed to build config from kubeconfig")
+		return nil, errors.Wrap(err, "loading kubeconfig")
 	}
-	return kubernetes.NewForConfig(config)
+	// using the kubelet kubeconfig, but with the iam-ra credentials file set
+	// if the node is running the iam-ra service, this will avoid starting a new session
+	// to make the kuberenetes api calls
+	config.AuthInfos["kubelet"].Exec.Env = []clientcmdapi.ExecEnvVar{
+		{Name: "AWS_SHARED_CREDENTIALS_FILE", Value: iamrolesanywhere.EksHybridAwsCredentialsPath},
```

Comment:
I think this method is used by debug, so not sure if this is the right place
what we can do is to accept in this methods closure options

the have another method in the hybrid package like `BuildKubeClient` and that method can call this one with the right option to se the aws creds file

then everywhere that needs a client (except debug) can call this new method. that way it's also obvious we are getting a kube client for the hybrid node case